qemu/tests/rng_hotplug.py

```diff
@@ -111,15 +126,22 @@ def stop_rngd(vm):
             hotplug_rng(vm, new_dev)
             dev_list.append(new_dev)
 
-        # Run test after hotplug
-        if test_after_hotplug and i == xrange(repeat_times)[-1]:
-            restart_rngd(vm)
-            error_context.context("Run %s after hotplug" % test_after_hotplug,
-                                  logging.info)
-            utils_test.run_virt_sub_test(test, params, env,
-                                         test_after_hotplug)
+        #run rng test after hotplug
+        check_and_run_subtest(rng_basic_test)
+
+        #run reboot/shutdown after hotplug
+        check_and_run_subtest(pm_test_after_plug)
+
+        #run rng test after reboot,skip followed test if pm_test_after_plug is shutdown
+        if vm.is_alive():
+            check_and_run_subtest(rng_basic_test)
+        else:
+            break
```

Comment:
Would it be better if we use `return` here instead of `break`?  
And any chance when pm_test_after_plug is NOT shutdown and vm is down ? Or it will be caught in subtest if error happens?

Reply:
return and break are both ok,I modify to return as you prefer.
And the unusual shutdown should be caught in previous subtest.